test/NuGet.Core.Tests/NuGet.Protocol.Tests/HttpHandlerResourceV3ProviderTests.cs

```diff
@@ -124,5 +128,119 @@ static IEnumerable<DelegatingHandler> GetDelegatingHandlers(HttpMessageHandler h
+            var callbackResult = clientHandler.ServerCertificateCustomValidationCallback.Invoke(null, null, null, SslPolicyErrors.RemoteCertificateChainErrors
+                & SslPolicyErrors.RemoteCertificateNameMismatch
+                & SslPolicyErrors.RemoteCertificateNotAvailable
+                & SslPolicyErrors.None);
+            callbackResult.Should().BeTrue();
+        }
+
+        [Fact]
+        public async Task TryCreate_WhenCertificateValidationIsEnabled_HandlerShouldBeNull()
+        {
+            // Arrange
+            Mock<IProxyCache> proxyCache = new();
+            proxyCache.Setup(pc => pc.GetProxy(It.IsAny<Uri>())).Returns((IWebProxy)null);
+            PackageSource packageSource = new(_testPackageSourceURL, "source")
+            {
+                DisableTLSCertificateValidation = false
+            };
+            SourceRepository sourceRepository = new(packageSource, Array.Empty<INuGetResourceProvider>());
+            HttpHandlerResourceV3Provider target = new(proxyCache.Object);
+
+            // Act
+            var result = await target.TryCreate(sourceRepository, CancellationToken.None);
+
+            // Assert
+            result.Item1.Should().BeTrue();
+            HttpHandlerResourceV3 resource = (HttpHandlerResourceV3)result.Item2;
+            resource.Should().NotBeNull();
+            HttpClientHandler clientHandler = resource.ClientHandler;
+
+            clientHandler.ServerCertificateCustomValidationCallback.Should().BeNull();
+        }
+
+        [Fact]
+        public async Task TryCreate_WhenCertificateValidationIsNotDisabled_ClientHandlerThrowsAnException()
+        {
+            // Arrange
+            TcpListenerServer server = new()
+            {
+                Mode = TestServerMode.InvalidTLSCertificate
+            };
+
+            Mock<IProxyCache> proxyCache = new();
+            proxyCache.Setup(pc => pc.GetProxy(It.IsAny<Uri>())).Returns((IWebProxy)null);
+            PackageSource packageSource = new(_testPackageSourceURL, "source");
+            SourceRepository sourceRepository = new(packageSource, Array.Empty<INuGetResourceProvider>());
+            HttpHandlerResourceV3Provider target = new(proxyCache.Object);
+            var result = await target.TryCreate(sourceRepository, CancellationToken.None);
+            HttpHandlerResourceV3 resource = (HttpHandlerResourceV3)result.Item2;
+            HttpClientHandler clientHandler = resource.ClientHandler;
+            var client = new HttpClient(clientHandler);
+
+            await server.ExecuteAsync(async uri =>
+            {
+                // Act & Assert
+                var exception = await Assert.ThrowsAsync<HttpRequestException>(async () => await client.GetAsync(uri));
+                return 0;
+            });
+        }
+
+        [Fact]
+        public async Task TryCreate_WhenCertificateValidationIsDisabled_ClientHandlerDoesNotThrowAnException()
```

Comment:
nit: Is there a more precise name?

```suggestion
        public async Task TryCreate_InvalidCertificateWithValidationDisabled_ClientHandlerDoesNotThrowAnException()
```